page_objects/products.list.page.js

```diff
@@ -1,31 +1,112 @@
+    }
+
+    getPollFountain() {
+        return new Button(element(by.css(pollFountainLocator)), "Poll, fountain button locator");
+    }
+
     getBuyOnHotlineCheckbox() {
         return new Checkbox(element(by.css(buyOnHotliheCheckboxLocator)), "Buy on Hotline checkbox");
     }
 
-    getAddToChurtButton() {
-        return new Button(element(by.css(addToChartButtonLocator)), "Add to chart button");
+    getAddToCartButton(itemNumber) {
+        return new Button(element(by.css(`.tile-viewbox > ul li:nth-of-type(${itemNumber}) .item-checkout`)), "Add to chart button");
+    }
+
+    getFilterItemCheckbox() {
+        return new Checkbox(element(by.css(filterItemLocator)), "Filter item checkbox");
     }
 
-    async checkCheckbox() {
-        await allure.createStep("Check Buy on Hotline chechbox", async () => {
+    getMinPriceInput() {
+        return new Input(element(by.css(minPriceInputLocator)), "Min price input");
+    }
+
+    getMaxPriceInput() {
+        return new Input(element(by.css(maxPriceInputLocator)), "Max price input");
+    }
+
+    getFoundItem() {
+        return new View(element(by.css(foundItemLocator)), "First found item in list");
+    }
+
+    getOkButton() {
+        return new Button(element(by.css(okButtonLocator)), "OK button");
+    }
+
+    getLoadAnimation() {
+        return new View(element(by.css(loadAnimationLocator)), "Loading animation");
+    }
+
+    async addToCart(itemNumber) {
+        await allure.createStep("Add item to cart", async () => {
+            await browser.actions().mouseMove(this.getDachaSad().getProtractorElement()).perform();
+            await this.getPollFountain().click();
+            await this.getPumps().click();
             await this.getBuyOnHotlineCheckbox().check();
+            await this.getAddToCartButton(itemNumber).click();
+        })();
+    }
+
+    async checkFilterCheckbox() {
+        await allure.createStep("Check filter item checkbox", async () => {
+            await this.getFilterItemCheckbox().check();
+        })();
+    }
+
+    async enterMinPrice(minPrice) {
+        await allure.createStep(`Enter min price - ${minPrice}`, async () => {
+            await this.getMinPriceInput().sendKeys(protractor.Key.DELETE);
+            await this.getMinPriceInput().sendKeys(minPrice);
         })();
     }
 
-    async clicAddToChart() {
-        await allure.createStep("Click Add to Chard button", async () => {
-            await this.getAddToChurtButton().click();
+    async enterMaxPrice(maxPrice) {
+        await allure.createStep(`Enter min price - ${maxPrice}`, async () => {
+            await this.getMaxPriceInput().sendKeys(protractor.Key.DELETE);
+            await this.getMaxPriceInput().sendKeys(protractor.Key.DELETE);
+            await this.getMaxPriceInput().sendKeys(protractor.Key.DELETE);
+            await this.getMaxPriceInput().sendKeys(protractor.Key.DELETE);
+            await this.getMaxPriceInput().sendKeys(protractor.Key.DELETE);
```

Comment:
move all `await this.getMaxPriceInput().sendKeys(protractor.Key.DELETE);` inside loop